src/vuex/store.js

```diff
@@ -178,33 +228,12 @@ const store = new Vuex.Store({
     SET_DISPLAY_ORIGIN(state, pos) {
       state.displayOrigin = pos;
     },
-    MOVE_PLAYER_UP(state) {
-      const pos = {
-        x: state.player.pos.x,
-        y: state.player.pos.y - 1,
-      };
-      state.player = Object.assign({}, state.player, {pos});
-    },
-    MOVE_PLAYER_DOWN(state) {
-      const pos = {
-        x: state.player.pos.x,
-        y: state.player.pos.y + 1,
-      };
-      state.player = Object.assign({}, state.player, {pos});
-    },
-    MOVE_PLAYER_LEFT(state) {
+    MOVE_PLAYER(state, delta) {
       const pos = {
-        x: state.player.pos.x - 1,
-        y: state.player.pos.y,
+        x: state.player.pos.x + delta[0],
+        y: state.player.pos.y + delta[1],
       };
-      state.player = Object.assign({}, state.player, {pos});
-    },
-    MOVE_PLAYER_RIGHT(state) {
-      const pos = {
-        x: state.player.pos.x + 1,
-        y: state.player.pos.y,
-      };
-      state.player = Object.assign({}, state.player, {pos});
+      state.player = Object.assign({}, state.player, { pos });
```

Comment:
Why are you assigning a copy of the player (with a new pos) to the player var? Why not just modify the player? `Object.assign(state.player, { pos });`

Reply:
I had been hit by a spate of immutability. But only for this mutation, not for others... :grimacing:

Reply:
Yeah, immutability is most useful in threaded environments, and only when an object is always immutable